workflow/controller/workflowpod.go

```diff
@@ -317,9 +317,16 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 		pod.Spec.InitContainers[i] = c
 	}
 
+	envVarTemplateValue := wfv1.MustMarshallJSON(tmpl)
+	templateSize := len(envVarTemplateValue)
+	if templateSize > 128000 {
```

Comment:
ARG_MAX is 131KB so we could allow a slightly higher number but there are other environment variables being added. 

```
docker run --entrypoint ""  -it quay.io/argoproj/argoexec:latest sh                       
/ # getconf ARG_MAX
131072
```

Reply:
We could loop over all the `envVars` and add them in a separate error message.

Reply:
@alexec is there any special way to note this as a breaking change and where would be the best place to document the new limit?

Reply:
I thought we were going to fix this by splitting the single ARG (max 130k) into two args (make 260k) which would fix the issue.

Reply:
Docs are still good mind you, so I approve of this.